pkg/ibm_plugin/sdk/tagging.go

```diff
@@ -42,20 +42,19 @@ func (c *CloudClient) attachTag(CRN *string, tags []string) error {
 
 	// attach tags with retires.
 	// retry mechanism improves stability and is needed due to possible temporary unavailability of resources, e.g. at time of creation.
-	maxAttempts := 10 // retries number to tag a resource
-	var err error
-	var result *globaltaggingv1.TagResults
-	for attempt := 1; attempt <= maxAttempts; attempt++ {
-		result, _, err = c.taggingService.AttachTag(attachTagOptions)
+	maxAttempts := 20    // retries number to tag a resource
+	latestResponse := "" // record latest response from inner scope
+	for attempt := 1; attempt <= maxAttempts; attempt += 1 {
+		result, latestResponse, _ := c.taggingService.AttachTag(attachTagOptions)
 		// tracking all responses from error prone tagging service
-		utils.Log.Printf("Tagging attempt %v: Error: %v", attempt, err)
+		utils.Log.Printf("Tagging attempt %v: Response: %+v", attempt, latestResponse)
```

Comment:
Hey @praveingk, Although it clutters the logs it is necessary, since this service doesn't always return an error when it fails, and when it does return an error, it doesn't always return a meaningful error message.

Reply:
@praveingk I've made changes to the tagging file, per your request, please review.
With regard to your questions:
- `latestResponse` is very useful, since it contains info that ideally should be part of the error returned, but isn't.
- From my extensive experience with the tagging service, I've come to think of it as a stochastic function. To reduce its probability of error we are running it multiple times. As I'm writing this message I'm also opening a ticket with cloud support after 20+ failed tagging attempts during my latest test run.